edx/edx-ora2
# feat: backend rubric reuse changes

**TL;DR -** Backend changes for rubric reuse

JIRA: [EDUCATOR-5752](https://openedx.atlassian.net/browse/EDUCATOR-5752)

**What changed?**

- Added a new mixin, RubricReuseMixin, that provides rubric-reuse-related functionality
- `get_other_course_ora_blocks` returns the list of all other ORA XBlocks in the course
-  added `get_other_ora_blocks_for_rubric_editor_context`, which internally calls `get_other_course_ora_blocks` and returns a list of dicts with block Display Name and Location
- `get_rubric` returns the rubric information for an ORA XBlock given a block locator
- modify `StudioMixin.get_editor_context` to call `get_other_ora_blocks_for_rubric_editor_context` if the feature flag is enabled, and return two new keys in the context dict: 
               - `'rubric_reuse_enabled'`: is the feature enabled?
               - `'other_course_openassessments'`: the return value of `get_other_ora_blocks_for_rubric_editor_context`, unless the flag is disabled, then an empty list.
- Added opaque-keys as a requirement

**Developer Checklist**


**Testing Instructions**

-I'm actually incredibly unsure!! I manually tested by adding the following lines to `studio_mixin.py:149` ( above `return { 'prompts':`)

```
        other_ora_blocks = self.get_other_course_ora_blocks()
        for other_ora_block in other_ora_blocks:
            logger.info(block.location)
            logger.info(self._get_rubric(other_ora_block.location))
```

Then, when you go to studio and edit an ORA, when you click the "edit" button you should see all other ORAs and rubrics in the logs

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
